x-pack/plugins/observability_solution/apm/public/components/app/onboarding/instructions_set.tsx

```diff
@@ -41,9 +42,18 @@ export function InstructionsSet({
 }) {
   const tabs = getTabs(instructions.instructionVariants);
 
-  const [selectedTab, setSelectedTab] = useState<string>(tabs[0].id);
+  const { hash } = useLocation();
+  const history = useHistory();
+  const preselectedTab = tabs.find(({ id }) => hash.slice(1) === id);
+  const [selectedTab, setSelectedTab] = useState<string>(
+    preselectedTab?.id ?? tabs[0].id
+  );
   const onSelectedTabChange = (tab: string) => {
     setSelectedTab(tab);
+    history.push({
```

Comment:
APM also has a [push helper](https://github.com/cauemarcondes/kibana/blob/f27d7da7e176b08833b16a00be44f7118074ea05/x-pack/plugins/observability_solution/apm/public/components/shared/links/url_helpers.ts#L53)

Reply:
Check out https://github.com/elastic/kibana/commit/1ec9265b42070afb074664b49c3e7f03565eecae and let me know if this looks better now.